ivadomed/inference.py

```diff
@@ -34,36 +40,61 @@ def onnx_inference(model_path, inputs):
     ort_outs = ort_session.run(None, ort_inputs)
     return torch.tensor(ort_outs[0])
 
-def get_preds(context: dict, model, fname_model: str, model_params: dict, cuda_available: bool, batch: dict):
+
+def get_preds(context: dict, path_model_file: str, model_params: dict, gpu_id: int, batch: dict) -> tensor:
     """Returns the predictions from the given model.
 
     Args:
         context (dict): configuration dict.
-        model: model used to obtain predictions.
-        fname_model (str): name of file containing model.
+        path_model_file (str): name of file containing model.
         model_params (dict): dictionary containing model parameters.
-        cuda_available (bool): true if cuda is available.
+        gpu_id (int): Number representing gpu number if available. Currently does NOT support multiple GPU segmentation.
         batch (dict): dictionary containing input, gt and metadata
 
-
     Returns:
-        Tensor: predictions from the model.
+        tensor: predictions from the model.
     """
+    # Define device
+    cuda_available = torch.cuda.is_available()
+    logger.debug(f"CUDA Status: {cuda_available}")
+    device = torch.device("cpu") if not cuda_available else torch.device("cuda:" + str(gpu_id))
```

Comment:
Is it to avoid the import of `utils` here? cf [this function](https://github.com/ivadomed/ivadomed/blob/master/ivadomed/utils.py#L117)

Reply:
Good spot! I totally didn't know about that function, we can totally import this. I had to refactor it from the earlier code as it was a bit confusing but if we have dedicated function. I will use that instead. Excellent!